Sources/Controllers/MyPlaces/TracksFilterDetailsViewController.swift

```diff
@@ -0,0 +1,766 @@
+                row.icon = .icCustomTrackLineMedium
+            }
+            row.iconTintColor = .iconColorDisruptive
+        } else if let itemInt = GpxUtilities.shared.parseColor(colorString: itemName) {
+            row.iconTintColor = colorFromRGB(Int(truncating: itemInt))
+        }
+    }
+    
+    private func getMeasureUnitType() -> MeasureUnitType {
+        if let rangeFilterType = rangeFilterType {
+            return rangeFilterType.trackFilterType.measureUnitType
+        }
+        
+        return .none
+    }
+    
+    private func updateAllFoldersSelection(add: Bool) {
+        func updateFolderSelection(from folder: DisplayFolderItem, add: Bool) {
+            if add {
+                if !selectedItems.contains(folder.key) {
+                    selectedItems.append(folder.key)
+                }
+            } else {
+                if let index = selectedItems.firstIndex(of: folder.key) {
+                    selectedItems.remove(at: index)
+                }
+            }
+            
+            for subfolder in folder.subfolders {
+                updateFolderSelection(from: subfolder, add: add)
+            }
+        }
+        
+        for folder in allFoldersListItems {
+            updateFolderSelection(from: folder, add: add)
+        }
+    }
+    
+    private func selectAllFolders() {
+        updateAllFoldersSelection(add: true)
+    }
+    
+    private func deselectAllFolders() {
+        updateAllFoldersSelection(add: false)
+    }
+    
+    private func checkIfAllFoldersSelected() {
+        let folderItemsToDisplay = isSearchActive ? filteredFoldersListItems : allFoldersListItems
+        let allFolderKeys = folderItemsToDisplay.flatMap { flattenFolder($0).map { $0.key } }
+        listFilterType?.isSelectAllItemsSelected = allFolderKeys.allSatisfy { selectedItems.contains($0) }
+    }
+    
+    private func flattenFolder(_ folderItem: DisplayFolderItem) -> [DisplayFolderItem] {
+        var result: [DisplayFolderItem] = [folderItem]
+        for subfolder in folderItem.subfolders {
+            result.append(contentsOf: flattenFolder(subfolder))
+        }
+        
+        return result
+    }
+    
+    func updateRangeValues() {
+        isBinding = true
+        if currentMaxValue > currentMinValue {
+            rangeSliderMinValue = Float(currentMinValue)
+            rangeSliderMaxValue = Float(currentMaxValue)
+            rangeSliderFromValue = Float(currentValueFrom)
+            rangeSliderToValue = Float(currentValueTo)
+            valueFromInputText = String(describing: Int(currentValueFrom))
+            valueToInputText = String(describing: Int(currentValueTo))
+            let mappedConstant = TracksSearchFilter.mapEOAMetricsConstantToMetricsConstants(OAAppSettings.sharedManager().metricSystem.get())
+            minFilterValueText = "\(decimalFormatter.string(from: NSNumber(value: Float(currentMinValue))) ?? "") \(getMeasureUnitType().getFilterUnitText(mc: mappedConstant))"
```

Comment:
Create a separate function for formatting minFilterValueText and maxFilterValueText